### PR TITLE
Add `<LineChart.HoverTrap />`

### LINKED ISSUE #30: Add `hover` trigger for Web
Currently, the touch handler works on web for tracking your current position. I think it would be cool to do the same with your mouse hover. I assume it should be somewhat easy. It's too bad RNGH doesn't have a `HoverGestureHandler`. But I assume there are many such solutions out there. We just need to calculate the mouse's position relative to the `PanGestureHandler` and use this to update the shared values.

### PR DESCRIPTION
**Pull request by nandorojo**
Close #30 

```tsx
<LineChart.CursorLine>
  <LineChart.HoverTrap />
</LineChart.CursorLine>
```

### CODE REVIEW

File: src/charts/candle/HoverTrap/index.web.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import { View, StyleSheet } from 'react-native';
+import { parse } from 'react-native-redash';
+
+import { LineChartDimensionsContext } from '../../line/Chart';
```
Comment (jxom) [inline src/charts/candle/HoverTrap/index.web.tsx:5]: Is this meant to reference the candle implementation?

Comment (nandorojo) [reply src/charts/candle/HoverTrap/index.web.tsx:5]: Oh whoops this was unfinished. We should just remove the candle one for this PR.

Comment (nandorojo) [reply src/charts/candle/HoverTrap/index.web.tsx:5]: I don’t actually use the candle atm so I haven’t tested there. But I assume the same logic should apply.

Comment (jxom) [reply src/charts/candle/HoverTrap/index.web.tsx:5]: Yep. I am happy to leave it out for candle for the moment. Do you think we could add some quick docs to the readme for this too?

Comment (nandorojo) [reply src/charts/candle/HoverTrap/index.web.tsx:5]: Yeah I can add some docs and fix the conflicts when I have a second soon.

also, I just realized that the readme has messed up tables currently (scroll down to see them)

**Comment by nandorojo** [comment]
Docs added.

**Pull request merged**